roles/pool/config-examples/pool-config-hosted-tp-example.toml

```diff
@@ -25,3 +25,4 @@ pool_signature = "Stratum v2 SRI Pool"
 #tp_address = "127.0.0.1:8442"
 # Hosted testnet TP 
 tp_address = "75.119.150.111:8442"
+tp_authority_public_key = "EguTM8URcZDQVeEBsM4B5vg9weqEUnufA8pm85fG4bZd"
```

Comment:
I think we use `3VANfft6ei6jQq1At7d8nmiZzVhBFS4CiQujdgim1ign` as a placeholder key everywhere.

Reply:
that is the default value of `authority_public_key` on the config file.

https://github.com/stratum-mining/stratum/blob/a0cc7913559412b6b9a5d891dc344368b4879e2d/roles/pool/config-examples/pool-config-hosted-tp-example.toml#L2

I thought we wanted `tp_authority_public_key` to hold a value that's coming from Bitcoin Core logs?
I got this value from:
```
2024-02-09T16:40:46Z Template Provider authority key: EguTM8URcZDQVeEBsM4B5vg9weqEUnufA8pm85fG4bZd
```

Reply:
It's randomly generated the first time you run the template provider. So it's going to be different for everyone.

Reply:
I plan to add support for importing a certificate later, so you'll be able to use an existing authority key. I.e. a pool might want that.

Reply:
Do we assume that for the communication pool <--> TP the authority is the TP itself?

Reply:
> ok but we shouldn't really use `3VANfft6ei6jQq1At7d8nmiZzVhBFS4CiQujdgim1ign` for `tp_authority_public_key`, should we?

You could also leave it empty in the example, since the user will have to fill it in.

> Do we assume that for the communication pool <--> TP the authority is the TP itself?

I don't want to assume that. It's probably true, but maybe the pool wants to use an external TP. Or maybe they run their own, but they don't want to bother signing a certificate - in that case they can just use the random self-signed certificate generated by Bitcoin Core.

Reply:
> I plan to add support for importing a certificate later, so you'll be able to use an existing authority key. I.e. a pool might want that.

that is an interesting idea, and brings a few questions to my mind.

basically, this TP certificate will change the workflow we have on the [Getting Started](https://stratumprotocol.org/getting-started/) tutorials. I can see a couple directions to go:
- user takes certificate that is automatically generated by the TP and manually edits the example pool config file
- user imports a pre-generated certificate into the TP, while the example pool config file already points at it by default.

IMO the metric we should aim for are the tradeoffs between:
- how intuitive and easy-flowing we allow the tutorials to be
- how we emphasize the security concerns about proper certificate management

Reply:
>> Do we assume that for the communication pool <--> TP the authority is the TP itself?

> I don't want to assume that. It's probably true, but maybe the pool wants to use an external TP. Or maybe they run their own, but they don't want to bother signing a certificate - in that case they can just use the random self-signed certificate generated by Bitcoin Core.

I agree with @Sjors on this one. Our baseline should be the most generic possible scenario.

Reply:
Updated the base branch.
After that @Sjors gives his consent on the rewriting of commits' history, you can rebase

Reply:
> user imports a pre-generated certificate

That won't work, because the certificate signs for the static key. And the static is randomly generated at startup. You'd also have to import that.

An easier solution IMO is to allow `tp_authority_public_key` to be empty. In that case we don't verify the certificate.

Reply:
for now I just added some comments to the config files with instructions on how to take the TP key

https://github.com/lorbax/stratum/pull/3/commits/10acde04ac376d35c5c52f2d8b5d281630823cc4

Reply:
given that right now the TP have random keys, we could add the possibility to leave the auth pub key for the TP in the pool set to None (void string).

So that if we have None we do not check the cert, if we have a key we check it. So now it will work, cause we leave it void and  when the TP can have static key we just change the config and we put there the TP key and it will do what is supposed to do.

Reply:
`None` makes sense, see #746.

Reply:
ok I'm adapting this PR accordingly

Reply:
You have `Initiator::new(authority_public_key)`. So you should change the type of `initiator.authority_public_key`[^1] to `Option<XOnlyPublicKey>` and pass an option also to the `new` function. So that when `authority_public_key` is `Some(authority_public_key)` you do the check otherwise you don't.


[^1]: I would call the field name `authority_public_key` rather `responder_authority_pk` cause `authority_public_key` is what we use in the spec, and for this cases stick impl_name = identity(spec_name) can be helpful.

Reply:
yeah that's the direction I was going for

but when `authority_public_key` is `None`, what do we return for `step_2()`?

Reply:
I still think that adding support for `None` can be done in a followup PR.

Reply:
@Sjors it will, but I asked this question to @Fi3 before you had made this suggestion so I'm just using the momentum